### PR TITLE
Fix device placement for GRPO attention mask in compute_loss

### PR DESCRIPTION
**Pull request by tgaddair**
Follow-up to #2708. Seems the attention_mask in the prompt_inputs can be CPU even when the rest of training is on GPU.

cc @qgallouedec

### CODE REVIEW

**Comment by ctjlewis** [comment]
good catch, cc @qgallouedec

**Comment by qgallouedec** [comment]
Surprising, since the device placement should be handled by 

```python
prompt_inputs = super()._prepare_inputs(prompt_inputs)
```

can you provide a code to reproduce?
